yamlpath/merger/mergerconfig.py

```diff
@@ -24,21 +24,28 @@
 class MergerConfig:
     """Config file processor for the Merger."""
 
-    def __init__(self, logger: ConsolePrinter, args: Namespace) -> None:
+    def __init__(
+            self,
+            logger: ConsolePrinter,
+            args: Namespace,
+            **kwargs: Any,
+    ) -> None:
         """
         Instantiate this class into an object.
 
         Parameters:
         1. logger (ConsolePrinter) Instance of ConsoleWriter or subclass
         2. args (dict) Default options for merge rules
+        3. kwargs (dict) Overrides for config values
 
         Returns:  N/A
         """
         self.log = logger
         self.args = args
-        self.config: Union[None, configparser.ConfigParser] = None
+        self.config: Optional[configparser.ConfigParser] = None
         self.rules: Dict[NodeCoords, str] = {}
         self.keys: Dict[NodeCoords, str] = {}
+        self.config_overrides: Dict[str, Any] = kwargs
```

Comment:
Down the road, other keyword arguments could be added, so consuming the whole of `kwargs` for this purpose is problematic.  Apart from that, `self.config_overrides` is never used outside of `self._load_config()`, which is called immediately after setting the instance variable.  To add some future-proofing to this and eliminate the otherwise unused instance variable, try this:

```python
override_keys: Dict[str, Any] = **kwargs.pop("keys", {})
override_rules: Dict[str, Any] = **kwargs.pop("rules", {})
config_overrides: Dict[str, Any] = {}

if len(override_keys):
    config_overrides["keys"] = override_keys
if len(override_rules):
    config_overrides["rules"] = override_rules

self._load_config(config_overrides)
```

or something like it.

Reply:
Implemented something similar. Let me know what you think. I wanted to avoid the pop since that's a mutable operation, not that I imagine anyone would notice.

Reply:
In team programming, there is a rule, "Be internally consistent."  The `pop` method is used throughout the entire rest of the code-base.  Avoiding it in one place is inconsistent.  Besides that, there's no benefit to preserving kwargs.  It's an ephemeral object generated by Python for the specific function call.  No one anywhere is looking to get it back.

Reply:
I appreciate you taking the time to explain your perspective. As an experienced developer, I do have a solid understanding of coding conventions and best practices. Though I agree with and will make your requested changes, in this case, I made a deliberate choice to handle the kwargs differently for the reasons I explained previously. Please don't assume my perspective results from lack of experience - I aim to make choices in the project's best interest.

Reply:
From my time in Japan, one of my favorite sayings translates to, "Even monkeys sometimes fall from trees."  In other words, even experts make mistakes from time to time.  I look forward to your updated contribution.